components/next_steps_view/steps/setup_preferences_step/setup_preferences_step.tsx

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+
+import UserSettingsModal from '../../../user_settings/modal';
+
+import {StepComponentProps} from '../../steps';
+
+import TextCardWithAction from '../text_card_with_action/text_card_with_action';
+
+export default function SetupPreferencesStep(props: StepComponentProps) {
+    const [shouldShowSettings, showSettings] = useState(false);
+    const onFinish = () => {
+        props.onFinish(props.id);
+    };
+    const onClick = () => {
+        showSettings(true);
+    };
+
+    return (
+        <>
+            <TextCardWithAction
+                cardBodyMessageId={'next_steps_view.preferenceSetup'}
+                cardBodyDefaultMessage={
+                    'You can change how you receive notifications, update your profile, customize display settings and more. Preferences can be accessed through the Main Menu.'
+                }
+                buttonMessageId={
+                    'next_steps_view.preferenceSetup.setPreferences'
+                }
+                buttonDefaultMessage={'Set Preferences'}
+                onClick={onClick}
+            />
+            {shouldShowSettings && (
+                <UserSettingsModal onHide={onFinish}/>
```

Comment:
This modal is hooked up to redux, so we should have it open with `openModal` as it is a better practice to make use of the `ModalController`

Reply:
I cannot for the life of me get the UserSettingsModal to call the onHide prop I pass it through `dialogProps` in openModal. I've tried both with connecting the component, and using the useDispatch hook. No matter what I do, my passed onHide function is never called. Will need some time to dig into this one more

Reply:
Let me know if I can be of some help :)

Reply:
Should be good in the next spinwick.